### PR TITLE
AIP-84 Migrate POST list Dag Runs(batch) endpoint to FastAPI 

### PR DESCRIPTION
**Pull request by rawwar**
related to #42701

### CODE REVIEW

File: tests/api_fastapi/core_api/routes/public/test_dag_run.py [hunk 1]
```diff
@@ -432,6 +469,333 @@ def test_invalid_state(self, test_client):
...
 
+class TestListDagRunsBatch:
+    @staticmethod
+    def parse_datetime(datetime_str):
+        return datetime_str.isoformat().replace("+00:00", "Z") if datetime_str else None
+
+    @staticmethod
+    def get_dag_run_dict(run: DagRun):
+        return {
+            "run_id": run.run_id,
```
Comment (rawwar) [inline tests/api_fastapi/core_api/routes/public/test_dag_run.py]: Once https://github.com/apache/airflow/pull/44305 is merged, I need to take changes and update tests

Comment (rawwar) [reply tests/api_fastapi/core_api/routes/public/test_dag_run.py]: Done

File: tests/api_fastapi/core_api/routes/public/test_dag_run.py [hunk 2]
```diff
@@ -432,6 +469,333 @@ def test_invalid_state(self, test_client):
...
+            ("id", DAG_RUNS_LIST),
+            ("state", [DAG1_RUN2_ID, DAG1_RUN1_ID, DAG2_RUN1_ID, DAG2_RUN2_ID]),
+            ("dag_id", DAG_RUNS_LIST),
+            ("logical_date", DAG_RUNS_LIST),
+            ("dag_run_id", DAG_RUNS_LIST),
+            ("start_date", DAG_RUNS_LIST),
+            ("end_date", DAG_RUNS_LIST),
+            ("updated_at", DAG_RUNS_LIST),
+            ("external_trigger", DAG_RUNS_LIST),
+            ("conf", DAG_RUNS_LIST),
```
Comment (pierrejeambrun) [inline tests/api_fastapi/core_api/routes/public/test_dag_run.py]: Be sure to test with `-` as well. Is that just a coincidence that all of the different orders return the same order ?

I'd like to see some different orders returned, maybe use `-dag_run_id`, `-logical_date` etc...

**Review by pierrejeambrun — APPROVED** [review]
Nice.

One minor comment, ready to merge after that.

**Review by pierrejeambrun — APPROVED** [review]

File: tests/api_fastapi/core_api/routes/public/test_dag_run.py (hunk 2 above)
Comment (pierrejeambrun) [reply tests/api_fastapi/core_api/routes/public/test_dag_run.py]: Thanks

**Pull request merged**